crates/bevy_reflect/src/list.rs

```diff
@@ -95,7 +95,7 @@ pub trait List: PartialReflect {
     fn iter(&self) -> ListIter;
 
     /// Drain the elements of this list to get a vector of owned values.
-    fn drain(self: Box<Self>) -> Vec<Box<dyn PartialReflect>>;
+    fn drain(&mut self) -> Vec<Box<dyn PartialReflect>>;
```

Comment:
While at it, doesn't it make sense to make drain take a range, like in the Vec::drain function? This would be a breaking change though

Reply:
I don't have a strong opinion here, but my slight preference is to keep the API minimal. I'm picturing something like a linked list may not want to provide draining across element indices. Also making the API even more different across every one of these traits could be confusing. Already the difference between the drains is a little confusing (though it makes sense with the nomenclature).